c_common/models/live_packet_gather/live_packet_gather.c

```diff
@@ -119,6 +156,27 @@ void flush_events(void) {
     buffer_index = 0;
 }
 
+//! \brief function to store provanence data elements into sdram
+void record_provanence_data(){
+    // Get the address this core's DTCM data starts at from SRAM
+    address_t address = data_specification_get_data_address();
+    // locate the provanence data region base address
+    address_t provanence_region_address =
+        data_specification_get_region(CONFIGURATION_REGION, address);
+    uint32_t provanence_data[NUMBER_OF_PROVANENCE_DATA_ELEMENTS];
+    // write provanence data into a array for memcpy
+    provanence_data[NUMBER_OF_LOSS_PACKETS_WITHOUT_PAYLOAD] =
+        number_of_over_flows_none_payload;
+    provanence_data[NUMBER_OF_LOSS_PACKETS_WITH_PAYLOAD] =
+        number_of_over_flows_payload;
+    // Copy provanence data into sdram region
+    memcpy(provanence_region_address, provanence_data,
+           SIZE_OF_PROVANENCE_ELEMENTS_IN_BYTES);
+    log_info("The provanence data consisting of %d lost packets without "
+             "payload and %d lost packets with payload.",
+             number_of_over_flows_none_payload, number_of_over_flows_payload);
+}
+
```

Comment:
As I said above, this seems overly circuitous! Either using a struct here or just copying straight into provanence_region_address would be be better.